library/clinical_rules.php

```diff
@@ -1491,6 +1584,37 @@ function buildPatientArrayPrimaryProviderBillingFacility($start, $batchSize, $on
  */
 function test_filter($patient_id, $rule, $dateTarget)
 {
+    /*
+    HR: test_filter() is called without first testing to see if a rule has any filters defined
+
+    test_filter() examines all of a rule's filters (and filter items) in one step
+
+    A "filter" for a given rule can contain multiple filter items, with each having an inclusion/exclusion flag
+    and a required/optional flag
+    The various filter evaluation "check" functions below will evaluate all filter items in a filter
+
+    The "check" functions below return:
+    true if all required filters (if any) pass, or if no required filters, and if any optional filters pass
+    'continue' if there are no required filters, and no optional filters pass
+    false if any required filters fail
+
+    If filters exist in one category and succeed, need to check other categories to see if required filters exist in those other categories as well
+
+    If filters in one category are all optional and do not succeed, try the next category
+
+    If inclusion filters in all categories are optional and do not succeed, return false (no need to check for exclusions)
+
+    If inclusion filters succeed, check for exclusions.
+    If no exclusions, return true
+    If exclusions exist and do not succeed, return true.
+    If exclusions succeed, return 'EXCLUDED'
+
+    If rule has no inclusion filters, but has exclusion filters, check the exclusion filters.
+    If exclusion filters succeed, return 'EXCLUDED'. If exclusion filters do not succeed, return true
+    (So rules do not have to have inclusion filters. If rule has only exclusion filters, and exclusion filters do not succeed, rule is applicable to patient)
+
+    if rule has no inclusion or exclusion filters, return false
```

Comment:
As discussed above, to keep current behavior would return `true` when no filters.

Reply:
btw, am reviewing the filter stuff first. Then will review target stuff after.